packages/mobile/src/exchange/saga.ts

```diff
@@ -400,16 +402,23 @@ function* withdrawCelo(action: WithdrawCeloAction) {
     const dollarAmount = yield call(celoToDollarAmount, amount)
     yield put(sendPaymentOrInviteSuccess(dollarAmount))
 
+    yield put(withdrawCeloSuccess())
+
     ValoraAnalytics.track(CeloExchangeEvents.celo_withdraw_completed, {
       amount: amount.toString(),
     })
   } catch (error) {
+    yield put(withdrawCeloFailed())
     Logger.error(TAG, 'Error withdrawing CELO', error)
     if (context?.id) {
       yield put(removeStandbyTransaction(context.id))
     }
 
-    yield put(showError(ErrorMessages.TRANSACTION_FAILED))
+    if (error.message === ErrorMessages.INCORRECT_PIN) {
+      yield put(showError(ErrorMessages.INCORRECT_PIN))
+    } else {
+      yield put(showError(ErrorMessages.TRANSACTION_FAILED))
+    }
```

Comment:
nit: ok to skip that now, but related to our previous discussion where we should migrate toward using events and avoid dispatching multiple actions sequentially, both `removeStandbyTransaction` and `showError` could be directly handled at the reducer level from the `withdrawCeloFailed` event (adding the necessary info to act on it of course).

Reply:
Yeah, you're right. I feel like this is a good time to start working on this, otherwise I'll probably forget about it next time I'm modifying something like this again. I changed it so that only one action is dispatched and that action is handled in the 3 reducers (the exchange one, the remove transaction one and the alerting one). Let me know what you think :)